hibernate-core/src/main/java/org/hibernate/tool/schema/internal/ColumnDefinitions.java

```diff
@@ -43,6 +43,21 @@ static boolean hasMatchingType(Column column, ColumnInformation columnInformatio
 		}
 	}
 
+	private static boolean hasSameColumnType(String metadataColumnTypeString, String columnInformationColumnTypeString) {
+		/*
+		 tokenize the metadata column type string, since that could include more than just the type information, such as e.g. 'unsigned',
+		 and we can't just check if it contains the other one, since that would fail with e.g. tinyint and int
+		 the column information type string, as per InformationExtractorJdbcDatabaseMetaDataImpl.addColumns (L160) should already
+		 only contain a single string
+		*/
+		for (String s : metadataColumnTypeString.split( " " ) ) {
```

Comment:
Just for the record, it seems the problem is that `ColumnInformation#getTypeName` only returns `tinyint` because it already tokenizes the type and throws away the `unsigned` part.
IMO we should rather fix this in `InformationExtractorJdbcDatabaseMetaDataImpl` to not throw away the `unsigned` part.